examples/sktime/flavor.py

```diff
@@ -0,0 +1,639 @@
+            ),
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+
+    with open(path, "rb") as pickled_model:
+        if serialization_format == SERIALIZATION_FORMAT_PICKLE:
+            return pickle.load(pickled_model)
+        elif serialization_format == SERIALIZATION_FORMAT_CLOUDPICKLE:
+            import cloudpickle
+
+            return cloudpickle.load(pickled_model)
+
+
+def _load_pyfunc(path):
+    """Load PyFunc implementation. Called by ``pyfunc.load_model``.
+
+    Parameters
+    ----------
+    path : str
+        Local filesystem path to the MLflow Model with the sktime flavor.
+
+    """
+    if os.path.isfile(path):
+        serialization_format = SERIALIZATION_FORMAT_PICKLE
+        _logger.warning("Loading procedure in older versions of MLflow using pickle.load()")
+    else:
+        try:
+            sktime_flavor_conf = _get_flavor_configuration(model_path=path, flavor_name=FLAVOR_NAME)
+            serialization_format = sktime_flavor_conf.get(
+                "serialization_format", SERIALIZATION_FORMAT_PICKLE
+            )
+        except MlflowException:
+            _logger.warning(
+                "Could not find sktime flavor configuration during model "
+                "loading process. Assuming 'pickle' serialization format."
+            )
+            serialization_format = SERIALIZATION_FORMAT_PICKLE
+
+        pyfunc_flavor_conf = _get_flavor_configuration(
+            model_path=path, flavor_name=pyfunc.FLAVOR_NAME
+        )
+        path = os.path.join(path, pyfunc_flavor_conf["model_path"])
+
+    return _SktimeModelWrapper(_load_model(path, serialization_format=serialization_format))
+
+
+class _SktimeModelWrapper:
+    def __init__(self, sktime_model):
+        self.sktime_model = sktime_model
+
+    def predict(self, X):
+        X = None if X.empty else X
+        raw_predictions = {}
+
+        if not hasattr(self.sktime_model, "pyfunc_predict_conf"):
+            raw_predictions[SKTIME_PREDICT] = self.sktime_model.predict(X=X)
+
+        else:
+            if not isinstance(self.sktime_model.pyfunc_predict_conf, dict):
+                raise MlflowException(
+                    f"Attribute {PYFUNC_PREDICT_CONF} must be of type dict.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if PYFUNC_PREDICT_CONF_KEY not in self.sktime_model.pyfunc_predict_conf:
+                raise MlflowException(
+                    f"Attribute {PYFUNC_PREDICT_CONF} must contain "
+                    f"a dictionary key {PYFUNC_PREDICT_CONF_KEY}.",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+
+            if isinstance(self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY], list):
+                predict_methods = self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY]
+                predict_params = False
+            elif isinstance(self.sktime_model.pyfunc_predict_conf[PYFUNC_PREDICT_CONF_KEY], dict):
```

Comment:
What are the values in this dict? Why support these two modes of both list and dict.keys()?

Reply:
(what I'm getting at is: "this is not intuitive or easy to follow. There should be dev comment notation here explaining what is going on and why a 'predict_params' value is set based on whether this is a list of effective keys or a dict with k:v entries")

Reply:
agree this needs more explanation in the code

essentially user can either define a list of predict methods without passing parameters (default parameters will be used), for example:

`["predict", "predict_interval"]`

or alternatively a dictionary with parameters, for example:

```
{
    "predict_method": {
        "predict": {},
        "predict_interval": {"coverage":  [0.8, 0.9]},
    }
}
```

Reply:
As mentioned above in my responses, I like your original design better :D

Reply:
not relevant anymore with revised interface